homeassistant/components/whois/helper.py

```diff
@@ -0,0 +1,64 @@
+"""A helper class that abstracts away the usage of external whois libraries."""
+
+from dataclasses import dataclass
+from typing import Any
+
+import whois
+
+
+@dataclass(kw_only=True)
+class Domain:
```

Comment:
Please take a look at the final code after all 3 planned PRs listed in the description. I believe it will help you understand why I decided to do it with a helper, instead of putting all the processing logic directly in HA classes.

https://github.com/mkrasowski/homeassistant-core/blob/replace-unmaintained-whois-library/homeassistant/components/whois/helper.py